modules/core/src/main/java/org/apache/synapse/message/store/impl/commons/MessageConverter.java

```diff
@@ -131,6 +132,10 @@ public static MessageContext toMessageContext(StorableMessage message,
                 axis2Ctx.setOperationContext(operationContext);
                 axis2Ctx.setAxisService(axisService);
                 axis2Ctx.setAxisOperation(axisOperation);
+            } else {
+                if (logger.isDebugEnabled()) {
```

Comment:
agreed that make sense. 
I suggested that thinking to reduce context switching. when looking into the source of logger wrapping with if seems more efficient
rough analysis
with if (logger.isDebugEnabled())      :   if ->  (X * context switch) -> if 
without if (logger.isDebugEnabled()) :   context switch -> if -> (X * context switch) -> if

leave unchanged. 
sorry for noise